rtCommon/bidsArchive.py

```diff
@@ -0,0 +1,571 @@
+"""-----------------------------------------------------------------------------
+
+bidsArchive.py
+
+Implements interacting with an on-disk BIDS Archive.
+
+-----------------------------------------------------------------------------"""
+import json
+import logging
+import numpy as np
+import os
+from typing import List
+
+from bids import BIDSLayout
+from bids.layout import parse_file_entities as bids_parse_file_entities
+from bids.layout.writing import build_path as bids_build_path
+import bids.config as bc
+import nibabel as nib
+
+from rtCommon.bidsCommon import (
+    BIDS_DIR_PATH_PATTERN,
+    getNiftiData,
+    isJsonPath,
+    isNiftiPath,
+)
+from rtCommon.bidsIncremental import BidsIncremental
+from rtCommon.errors import StateError, ValidationError
+from rtCommon.imageHandling import readNifti
+
+# Silence future warning
+bc.set_option('extension_initial_dot', True)
+
+logger = logging.getLogger(__name__)
+
+
+class BidsDataset:
+    def __init__(self, path: str):
+        self.name = os.path.basename(os.path.normpath(path))
+        # logger.debug("Loading dataset \"%s\" from: %s", self.name, path)
+        self.data = BIDSLayout(path)
+        # logger.debug("Dataset info: %s", self.data)
+
+    def __str__(self):
+        return str(self.data)
+
+    @staticmethod
+    def _stripRoot(relPath: str) -> str:
+        """
+        Strips a leading / from the path, preventing paths defined relative to
+        dataset root (/sub-01/ses-01) from being interpreted as being relative
+        to the root of the filesystem
+        """
+        if len(relPath) >= 1 and relPath[0] == "/":
+            return relPath[1:]
+        else:
+            return relPath
+
+    def absPathFromRelPath(self, relPath: str) -> str:
+        """
+        Makes an absolute path from the relative path within the dataset.
+        """
+        return os.path.join(self.data.root, self._stripRoot(relPath))
+
+    def fileExists(self, relPath: str) -> bool:
+        # search for matching BIDS file in the dataset
+        relPath = self._stripRoot(relPath)
+        result = self.data.get_file(self._stripRoot(relPath))
+        if result is not None:
+            return True
+
+    def dirExists(self, relPath: str) -> bool:
+        # search for matching directory in the dataset
+        return os.path.isdir(self.absPathFromRelPath(relPath))
+
+    def pathExists(self, path: str) -> bool:
+        return self.fileExists(path) or self.dirExists(path)
+
+    def findImage(self, path: str) -> nib.Nifti1Image:
```

Comment:
The function name seems a little unintuitive. My initial impression is it would return a path given an image name. Maybe getImage?